lib/zzhc/zzhc_port.c

```diff
@@ -0,0 +1,191 @@
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(zzhc_port, CONFIG_ZZHC_LOG_LEVEL);
+
+#include <zephyr.h>
+#include <base64.h>
+#include <secure_services.h>
+#include <json.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include "zzhc_internal.h"
+
+#define AT_PARAMS_MAX     11        /** Max. # of AT-params to parse */
+#define FICR_REV_OFFSET   0x134
+
+/** Cached storage of ICCID. */
+static char ueiccid[ICCID_LEN];
+
+/** Pointer to settings_handler. */
+static struct settings_handler settings_fn = {0};
+
+/** Name of settings */
+static char const settings_name[] = "zzhc";
+
+/**@brief Structure to store decoded json object. */
+struct rsp_obj {
+	const char *resultCode;
+	const char *resultDesc;
+};
+
+/**@brief Descriptor for json parser. */
+static const struct json_obj_descr rsp_desc[] = {
+	JSON_OBJ_DESCR_PRIM(struct rsp_obj, resultCode, JSON_TOK_STRING),
+	JSON_OBJ_DESCR_PRIM(struct rsp_obj, resultDesc, JSON_TOK_STRING),
+};
+
+int zzhc_read_chip_rev(u32_t *i)
+{
+	u32_t ficr_addr = NRF_FICR_S_BASE + FICR_REV_OFFSET;
+
+	return spm_request_read(i, ficr_addr, sizeof(u32_t));
+}
+
+int zzhc_load_iccid(char *iccid_buf, int buf_len)
+{
+	memcpy(iccid_buf, ueiccid, buf_len);
+	return 0;
+}
+
+int zzhc_base64(unsigned char *dst, size_t dlen, const unsigned char *src,
+	size_t slen)
+{
+	int i = 0, rc;
+
+	if (dlen == 0 || slen == 0) {
+		return 0;
+	}
+
+	rc = base64_encode(dst, dlen, &i, src, slen);
+	return (rc == 0) ? i : 0;
+}
+
+/**@brief Callback when settings_load() is called. */
+static int settings_set(const char *key, size_t len,
+		settings_read_cb read_cb, void *cb_arg)
+{
+	int rc;
+
+	if (strcmp(key, "iccid") == 0) {
+		rc = read_cb(cb_arg, ueiccid, ICCID_LEN);
+		if (rc < 0) {
+			return rc;
+		}
+	}
+
+	return 0;
+}
+
+int zzhc_ext_init(struct zzhc *ctx)
+{
+	int rc;
+
+	memset(ueiccid, 0, ICCID_LEN);
+
+	ctx->sem = zzhc_malloc(sizeof(struct k_sem));
+	if (ctx->sem == NULL) {
+		return -ENOBUFS;
+	}
+
+	ctx->at_list = zzhc_malloc(sizeof(struct at_param_list));
+	if (ctx->at_list == NULL) {
+		return -ENOBUFS;
+	}
+
+	/* Initialize AT-command list, which is used in parser */
+	rc = at_params_list_init(ctx->at_list, AT_PARAMS_MAX);
+	if (rc) {
+		LOG_DBG("at_params_list_init() = %d", rc);
+		return rc;
+	}
+
+	/* Initialize settings (run-once). */
+	if (settings_fn.name == NULL) {
+		rc = settings_subsys_init();
+		if (rc) {
+			LOG_DBG("settings_subsys_init() = %d", rc);
+			return rc;
+		}
+
+		settings_fn.name  = (char *)settings_name;
+		settings_fn.h_set = settings_set;
+		rc = settings_register(&settings_fn);
+		if (rc) {
+			LOG_DBG("settings_register() = %d", rc);
+			return rc;
+		}
+	}
+
+	/* Read from storage. */
+	rc = settings_load();
+	if (rc) {
+		return rc;
+	}
+
+	return 0;
+}
+
+void zzhc_ext_uninit(struct zzhc *ctx)
+{
+	zzhc_free(ctx->sem);
+	at_params_list_free(ctx->at_list);
+	zzhc_free(ctx->at_list);
+}
+
+bool zzhc_check_http_payload(struct zzhc *ctx)
+{
+	int rc;
+	bool resultCode, resultDesc = false;
```

Comment:
Same here, snake_case please.

Reply:
To be discussed [here](https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1340#discussion_r343123823).